alpha/alpha-core/src/main/java/org/apache/servicecomb/saga/alpha/core/EventScanner.java

```diff
@@ -109,6 +113,19 @@ private void updateCompensationStatus(TxEvent event) {
     markSagaEnded(event);
   }
 
+  private void abortTimeoutEvent() {
+    eventRepository.findFirstTimeoutEventByIdGreaterThan(nextTimeoutEventId)
+    .ifPresent((TxEvent event) -> {
+      log.info("Found timeout event {}", event);
+      nextTimeoutEventId = event.id();
+      eventRepository.save(toTxAbortedEvent(event));
+    });
+  }
+
+  private void updateTransactionStatus() {
+    eventRepository.findFirstAbortedGlobalTransaction().ifPresent(this::markGlobalTxEnd);
```

Comment:
why mark end without any compensation?

Reply:
timeout event with only SagaStartedEvent or only SagaStartedEvent and TxStartedEvent doesn't have anything to compensate, but currently there's no way to mark the end of such tx